src/components/Modal/Modal.tsx

```diff
@@ -211,4 +219,10 @@ export const Modal: React.FunctionComponent<ModalProps> & {
   );
 };
 
+function isRef(
+  ref: React.RefObject<HTMLElement> | React.ReactElement,
+): ref is React.RefObject<HTMLElement> {
+  return Object.prototype.hasOwnProperty.call(ref, 'current');
```

Comment:
Such a nice way to do a type guard 👍 

Just out of curiosity, does `ref.hasOwnProperty('current')` not work on react elements?

Reply:
I think it would but we have this rule enabled: https://eslint.org/docs/rules/no-prototype-builtins